controllers/storagecluster/generate.go

```diff
@@ -136,19 +136,40 @@ func generateNameForCephRbdMirror(initData *ocsv1.StorageCluster) string {
 
 // generateCephReplicatedSpec returns the ReplicatedSpec for the cephCluster
 // based on the StorageCluster configuration
-func generateCephReplicatedSpec(initData *ocsv1.StorageCluster, poolType string) cephv1.ReplicatedSpec {
+func generateCephReplicatedSpec(initData *ocsv1.StorageCluster, poolType string, storageType string) cephv1.ReplicatedSpec {
 	crs := cephv1.ReplicatedSpec{}
 
 	crs.Size = getCephPoolReplicatedSize(initData)
 	crs.ReplicasPerFailureDomain = uint(getReplicasPerFailureDomain(initData))
-	//lint:ignore ST1017 required to compare it directly
-	if "data" == poolType {
-		crs.TargetSizeRatio = .49
+	if poolType == poolTypeData {
+		crs.TargetSizeRatio = getTargetSizeRatio(initData, storageType)
 	}
 
 	return crs
 }
 
+func getTargetSizeRatio(initData *ocsv1.StorageCluster, storageType string) float64 {
+	defaultTargetSizeRatio := 0.49
+	var definedRatio float64
+
+	switch storageType {
+	case storageTypeBlock:
+		definedRatio = initData.Spec.ManagedResources.CephBlockPools.PoolSpec.Replicated.TargetSizeRatio
```

Comment:
Actually, the PoolSpec.TargetSizeRatio was deprecated in rook a long time ago, but we have kept it for backward compatibility. Instead, let's use the parameters:
```suggestion
		definedRatio = initData.Spec.ManagedResources.CephBlockPools.PoolSpec.Parameters["target_size_ratio"]
```

Reply:
Ack, @travisn Even though the field was deprecated there doesn't seem to any comment/kubebuilder warning in rook suggesting that. I think that can be added in rook APIs.

Reply:
@travisn Another question if both Replicated.TargetSizeRatio & Parameters["target_size_ratio"] are specified then which one has higher priority?

Reply:
I see the Replicated.TargetSizeRatio is given precedence ahead of the parameter.
https://github.com/rook/rook/blob/3fcc7682f2c3b02f6c9fed2184665ac6f0ce945b/pkg/daemon/ceph/client/pool.go#L296
And I also see the Replicated.TargetSizeRatio field being used here to determine if TragetSizeRatio is defined or not here https://github.com/rook/rook/blob/master/pkg/apis/ceph.rook.io/v1/pool.go#L39

Reply:
Yes, it's true the targetSizeRatio would have higher precedence, it was just implemented that way to easily support changing the implementation to use the parameters, and maintain backward compatibility. But you're right, we are long overdue to mark it as deprecated, it has been like this for many releases.

Reply:
With current rook behaviour we can't shift to parameters right now as it is, As if we move to the parameters the values left on the replicated.TargetSizeRatio would still have precedence.

Reply:
Oh I see now the point. Actually, just go ahead and keep using the replicated.TargetSizeRatio. We can keep Rook behavior the same and not bother to deprecate the setting, and it will keep working smoothly for downstream with your current approach.